### PR TITLE
Add HTTP status codes to PGRST errors

### PR DESCRIPTION
**Pull request by laurenceisla**
Adds HTTP status codes to the PGRST errors. Also removes deprecated errors and adds missing ones.

### CODE REVIEW

File: docs/errors.rst [hunk 1]
```diff
@@ -148,82 +148,89 @@ Group 1 - Api Request
...
++---------------+-------------+-------------------------------------------------------------+
+| .. _pgrst116: | 406         | More than 1 or no items where returned when requesting      |
+|               |             | a singular response. See :ref:`singular_plural`.            |
+| PGRST116      |             |                                                             |
++---------------+-------------+-------------------------------------------------------------+
+| .. _pgrst117: | 405         | The HTTP verb used in the request in not supported.         |
+|               |             |                                                             |
+| PGRST117      |             |                                                             |
++---------------+-------------+-------------------------------------------------------------+
+| .. _pgrst118: | 400         | Could not order the result using the related table because  |
```
Comment (laurenceisla) [inline docs/errors.rst:222]: I have a doubt about this error. It is also used when a column is not found when using `column`. These are [the lines of code](https://github.com/PostgREST/postgrest/blob/eb3f7c696a5da8c023ec090eb16fa05c87adf339/src/PostgREST/Error.hs#L214-L218).

Comment (steve-chavez) [reply docs/errors.rst:222]: @laurenceisla Yes, that needs to be corrected. It slipped by when reviewing https://github.com/PostgREST/postgrest/pull/2542

File: docs/errors.rst [hunk 2]
```diff
@@ -125,21 +125,21 @@ Group 0 - Connection
...
-| PGRST001      |                                                             |
-+---------------+-------------------------------------------------------------+
-| .. _pgrst002: | Could not connect with the database when building the       |
-|               | :ref:`schema_cache` due to the PostgreSQL service not       |
-| PGRST002      | running.                                                    |
-+---------------+-------------------------------------------------------------+
++---------------+-------------+-------------------------------------------------------------+
+| Code          | HTTP status | Description                                                 |
++===============+=============+=============================================================+
+| .. _pgrst000: | 503, 504    | Could not connect with the database due to an incorrect     |
```
Comment (steve-chavez) [inline docs/errors.rst:131]: Hm, IIRC `504` was added specially to handle the `db-pool-acquisition-timeout` error. Could we have a different PGRST error code for this case?

File: docs/errors.rst [hunk 3]
```diff
@@ -125,21 +125,25 @@ Group 0 - Connection
...
+|               |             | error.                                                      |
+| PGRST001      |             |                                                             |
++---------------+-------------+-------------------------------------------------------------+
+| .. _pgrst002: | 503         | Could not connect with the database when building the       |
+|               |             | :ref:`schema_cache` due to the PostgreSQL service not       |
+| PGRST002      |             | running.                                                    |
++---------------+-------------+-------------------------------------------------------------+
+| .. _pgrst003: | 503         | The request time exceeded the timeout specified in          |
+|               |             | :ref:`db-pool-acquisition-timeout`.                         |
+| PGRST003      |             |                                                             |
```
Comment (laurenceisla) [inline docs/errors.rst:145]: Should be merged after this config is added to the docs, I'll update with the PR# when it's ready.

Comment (laurenceisla) [reply docs/errors.rst:145]: Configuration parameter added in PR #591.

File: docs/errors.rst (hunk 1 above)
Comment (laurenceisla) [reply docs/errors.rst:226]: Fixed in https://github.com/PostgREST/postgrest/pull/2648

**Review by steve-chavez — APPROVED** [review]

**Pull request merged**